packages/openfeature-web-provider/src/ConfidenceWebProvider.e2e.test.ts

```diff
@@ -1,106 +1,114 @@
+
+OpenFeature.setLogger({
+  error: jest.fn(),
+  warn: jest.fn(),
+  info: jest.fn(),
+  debug: jest.fn(),
+});
+function createProvider(options: Partial<ConfidenceWebProviderOptions> = {}): Provider {
+  return createConfidenceWebProvider({
+    clientSecret: 'RxDVTrXvc6op1XxiQ4OaR31dKbJ39aYV',
+    timeout: 1000,
+    ...options,
+  });
+}
+describe('ConfidenceWebProvider E2E tests', () => {
+  describe('initialize fail', () => {
+    beforeEach(async () => {
+      await expect(OpenFeature.setProviderAndWait(createProvider({ timeout: 0 }))).rejects.toThrow(
+        'This operation was aborted',
+      );
     });
-    const providerReadyPromise = new Promise<void>(resolve => {
-      OpenFeature.addHandler(ProviderEvents.Ready, () => {
-        resolve();
-      });
-    }).then(() => {
-      return OpenFeature.setContext({
-        targetingKey: 'test-a', // control
-      });
+    afterEach(() => OpenFeature.clearProviders());
+
+    it('should have error status', () => {
+      const client = OpenFeature.getClient();
+      expect(client.providerStatus).toBe(ClientProviderStatus.ERROR);
     });
 
-    OpenFeature.setProvider(confidenceProvider);
+    it('should resolve default values', () => {
+      const client = OpenFeature.getClient();
 
-    return providerReadyPromise;
+      expect(client.getStringDetails('web-sdk-e2e-flag.str', 'default')).toEqual({
+        errorCode: 'PROVIDER_NOT_READY',
+        flagKey: 'web-sdk-e2e-flag.str',
+        flagMetadata: {},
+        reason: 'ERROR',
+        value: 'default',
+      });
+    });
   });
 
-  it('should return defaults after the timeout', async () => {
-    const confidenceProvider = createConfidenceWebProvider({
-      region: 'eu',
-      clientSecret: 'RxDVTrXvc6op1XxiQ4OaR31dKbJ39aYV',
-      timeout: 0,
+  describe('initialize success', () => {
+    beforeEach(async () => {
+      OpenFeature.setContext({ targetingKey: 'test-a' }); // control
+      await OpenFeature.setProviderAndWait(createProvider({ timeout: 1000 }));
     });
+    afterEach(() => OpenFeature.clearProviders());
 
-    await confidenceProvider.onContextChange!({}, { targetingKey: 'user-a' });
-
-    const flag = confidenceProvider.resolveStringEvaluation(
-      'web-sdk-e2e-flag.str',
-      'default',
-      { targetingKey: 'user-a' },
-      {
-        info: jest.fn(),
-        warn: jest.fn(),
-        error: jest.fn(),
-        debug: jest.fn(),
-      },
-    );
-
-    expect(flag.value).toEqual('default');
-  });
+    it('should have ready status', () => {
+      const client = OpenFeature.getClient();
+      expect(client.providerStatus).toBe(ProviderStatus.READY);
+    });
 
-  it('should resolve a boolean e2e', async () => {
-    const client = OpenFeature.getClient();
+    it('should resolve a boolean e2e', async () => {
+      const client = OpenFeature.getClient();
 
-    expect(client.getBooleanValue('web-sdk-e2e-flag.bool', true)).toBeFalsy();
```

Comment:
Changed this assertions to be more strict. `toBeFalsy` just means it's something that converts to `false`, we should want it to be an actual `false`